### PR TITLE
New gamemode Cult of flesh and antag Cultist of flesh

### PR DESCRIPTION
**Pull request by VigersRay**
## About the PR

I don’t even know what to write here, below there is a lot of information about what will be added and video materials. I hope you will help improve this mode and fix all the errors.
Formally, both the mode and the antagonist have long been made and have been tested for a month of gameplay.

**Media**

Preview:
Screenshot_69
image
image
More info:
[Brief description of the cult of the flesh.docx](https://github.com/space-wizards/space-station-14/files/11593622/Brief.description.of.the.cult.of.the.flesh.docx)
[Cult of Flesh.txt](https://github.com/space-wizards/space-station-14/files/11593639/Cult.of.Flesh.txt)
Video review:
https://youtu.be/AXkrHTibrHM

Gameplay on RU server:
https://youtu.be/N0OE78yHoPE

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl: ViggersRay
- add: Added cult of flesh gamemode
- add: Added cultist of flesh antag
- add: Added leader cultist of flesh antag
- add: Added more flesh mobs
- add: Added collective mind

### CODE REVIEW

**Comment by ficcialfaint** [comment]
CL has to include a verb, e.g.: 
❌ - add: Cool thing
✅ - add: Added cool thing

**Comment by asperger-sind** [comment]
Some, primarily mob, sprites look kinda out of place when compared to the rest of the graphics, as well as having animation cycles which most mobs lack.

**Comment by VigersRay** [comment]
> Some, primarily mob, sprites look kinda out of place when compared to the rest of the graphics, as well as having animation cycles which most mobs lack.


Well, I guess something will do with rewriting the mobs. However, the dragon does have animations.

**Comment by HoidC** [comment]
> Some, primarily mob, sprites look kinda out of place when compared to the rest of the graphics, as well as having animation cycles which most mobs lack.

I don't think they look too out of place - they look good. Animations make them look a lot more grotesque and arguably horrifying than the same old static mobs with single animations that we have currently.

**Comment by OctoRocket** [comment]
I love this so much. A unique spin on the classic cult theme with great graphics!

File: Resources/Textures/Mobs/Aliens/FleshCult/flesh_cult_abom.rsi/meta.json [hunk 1]
```diff
@@ -0,0 +1,56 @@
+{
+  "version": 1,
+  "license": "CC-BY-SA-3.0",
+  "copyright": "",
```
Comment (ficcialfaint) [inline Resources/Textures/Mobs/Aliens/FleshCult/flesh_cult_abom.rsi/meta.json:4]: why is it empty x2

File: Resources/Textures/Interface/Alerts/flesh_point.rsi/meta.json [hunk 2]
```diff
@@ -0,0 +1,62 @@
+{
+  "version": 1,
+  "license": "CC-BY-SA-3.0",
+  "copyright": "Created by ",
```
Comment (ficcialfaint) [inline Resources/Textures/Interface/Alerts/flesh_point.rsi/meta.json:4]: why is it empty

File: Resources/Textures/Mobs/Aliens/FleshCult/flesh_cult_abom.rsi/meta.json [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+          "name": "crit",
+          "directions": 1,
+          "delays": [
+              [
+                  0.3,
+                  0.3,
+                  0.3,
+                  0.3
+              ]
+          ]
```
Comment (ficcialfaint) [inline Resources/Textures/Mobs/Aliens/FleshCult/flesh_cult_abom.rsi/meta.json:53]: make the tabs the same for all states

**Comment by Maksim22848** [comment]
Ну наконец что-то реально крутое добавится

File: Resources/Textures/Mobs/Aliens/FleshCult/flesh_cult_mobs.rsi/meta.json [hunk 4]
```diff
@@ -0,0 +1,155 @@
+{
+    "version": 1,
+    "size": {
+        "x": 32,
+        "y": 32
+    },
+    "license": "CC-BY-SA-3.0",
+    "copyright": "",
```
Comment (ficcialfaint) [inline Resources/Textures/Mobs/Aliens/FleshCult/flesh_cult_mobs.rsi/meta.json:8]: why is it empty x3

File: Resources/Textures/Mobs/Aliens/FleshCult/flesh_cult_pudge.rsi/meta.json [hunk 5]
```diff
@@ -0,0 +1,56 @@
+﻿{
+  "version": 1,
+  "license": "CC-BY-SA-3.0",
+  "copyright": "",
```
Comment (ficcialfaint) [inline Resources/Textures/Mobs/Aliens/FleshCult/flesh_cult_pudge.rsi/meta.json:4]: why is it empty x4

File: Resources/Textures/Mobs/Aliens/FleshCult/flesh_cult_pudge.rsi/meta.json [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+          "name": "crit",
+          "directions": 1,
+          "delays": [
+              [
+                  0.3,
+                  0.3,
+                  0.3,
+                  0.3
+              ]
+          ]
```
Comment (ficcialfaint) [inline Resources/Textures/Mobs/Aliens/FleshCult/flesh_cult_pudge.rsi/meta.json:53]: tabulation

File: Resources/Textures/Objects/Misc/flesh.rsi/meta.json [hunk 7]
```diff
@@ -0,0 +1,38 @@
+﻿{
+    "version": 1,
+    "license": "CC-BY-SA-3.0",
+    "copyright": "https://github.com/tgstation/tgstation/commit/15bf91049e33979a855995579b48592e34bcdd8c",
```
Comment (ficcialfaint) [inline Resources/Textures/Objects/Misc/flesh.rsi/meta.json:4]: Make it like `Taken from tgstation at <link>`

File: Resources/Textures/Objects/Tools/flesh_claw.rsi/meta.json [hunk 8]
```diff
@@ -0,0 +1,22 @@
+{
+  "version": 1,
+  "license": "CC-BY-SA-3.0",
+  "copyright": "",
```
Comment (ficcialfaint) [inline Resources/Textures/Objects/Tools/flesh_claw.rsi/meta.json:4]: why is it empty x5

File: Resources/Textures/Objects/Weapons/Guns/Basic/flesh_spike_hand.rsi/meta.json [hunk 9]
```diff
@@ -0,0 +1,22 @@
+{
+  "version": 1,
+  "license": "CC-BY-SA-3.0",
+  "copyright": "",
```
Comment (ficcialfaint) [inline Resources/Textures/Objects/Weapons/Guns/Basic/flesh_spike_hand.rsi/meta.json:4]: why is it empty x?

File: Resources/Textures/Objects/Weapons/Guns/Projectiles/flesh_toxic.rsi/meta.json [hunk 10]
```diff
@@ -0,0 +1,14 @@
+{
+  "version": 1,
+  "license": "CC-BY-NC-4.0",
+  "copyright": "",
```
Comment (ficcialfaint) [inline Resources/Textures/Objects/Weapons/Guns/Projectiles/flesh_toxic.rsi/meta.json:4]: Why is it empty + why did you decide to change the license?

File: Resources/Textures/Objects/Weapons/Guns/Projectiles/spike.rsi/meta.json [hunk 11]
```diff
@@ -0,0 +1,14 @@
+{
+  "version": 1,
+  "license": "CC-BY-SA-3.0",
+  "copyright": "",
```
Comment (ficcialfaint) [inline Resources/Textures/Objects/Weapons/Guns/Projectiles/spike.rsi/meta.json:4]: empty

File: Resources/Textures/Objects/Weapons/Melee/flesh_blade.rsi/meta.json [hunk 12]
```diff
@@ -0,0 +1,22 @@
+{
+  "version": 1,
+  "license": "CC-BY-SA-3.0",
+  "copyright": "Created by discord:",
```
Comment (ficcialfaint) [inline Resources/Textures/Objects/Weapons/Melee/flesh_blade.rsi/meta.json:4]: semi-empty

File: Resources/Textures/Objects/Weapons/Melee/flesh_fist.rsi/meta.json [hunk 13]
```diff
@@ -0,0 +1,22 @@
+{
+  "version": 1,
+  "license": "CC-BY-SA-3.0",
+  "copyright": "Created by discord:",
```
Comment (ficcialfaint) [inline Resources/Textures/Objects/Weapons/Melee/flesh_fist.rsi/meta.json:4]: above

File: Resources/Textures/Structures/Walls/flesh.rsi/meta.json [hunk 14]
```diff
@@ -0,0 +1,46 @@
+{
+    "version": 1,
+    "size": {
+        "x": 32,
+        "y": 32
+    },
+    "license": "CC-BY-SA-3.0",
+    "copyright": "",
```
Comment (ficcialfaint) [inline Resources/Textures/Structures/Walls/flesh.rsi/meta.json:8]: empty

File: Resources/Textures/Structures/flesh_heart.rsi/meta.json [hunk 15]
```diff
@@ -0,0 +1,31 @@
+{
+    "version":1,
+    "size":{
+        "x":100,
+        "y":102
+    },
+    "license":"CC-BY-SA-3.0",
+    "copyright":"",
```
Comment (ficcialfaint) [inline Resources/Textures/Structures/flesh_heart.rsi/meta.json:8]: empty

**Review by ficcialfaint — CHANGES_REQUESTED** [review]

File: Content.Server/Flesh/FleshGrowth/SpreaderFleshSystem.cs [hunk 16]
```diff
@@ -0,0 +1,207 @@
...
+    {
+        if (EntityManager.TryGetComponent<SpreaderFleshComponent>(ent, out _))
+            return true;
+
+        if (!EntityManager.TryGetComponent<AirtightComponent>(ent, out var airtight))
+            return false;
+
+        // var oppositeDir = dir.AsDir().GetOpposite().ToAtmosDirection();
+
+        // return airtight.AirBlocked && airtight.AirBlockedDirection.IsFlagSet(oppositeDir);
```
Comment (ficcialfaint) [inline Content.Server/Flesh/FleshGrowth/SpreaderFleshSystem.cs:203]: Should be removed I think, if it's going to be used in the future or sum

File: Resources/Textures/Objects/Misc/flesh.rsi/meta.json (hunk 7 above)
Comment (VigersRay) [reply Resources/Textures/Objects/Misc/flesh.rsi/meta.json:4]: The original code was like this

Comment (ficcialfaint) [reply Resources/Textures/Objects/Misc/flesh.rsi/meta.json:4]: Wdym the original? Did you take it from somewhere?

Comment (VigersRay) [reply Resources/Textures/Objects/Misc/flesh.rsi/meta.json:4]: this is a copy of the floor from the flesh anomaly

**Comment by GoodWheatley** [comment]
Weren’t these sprites taken from another game? I remember you mentioning that during a discussion on Discord about them.

File: Resources/Audio/Ambience/Objects/license.txt [hunk 17]
```diff
@@ -8,3 +8,4 @@ drain.ogg - https://freesound.org/people/PhreaKsAccount/sounds/46266/ - CC-BY-3.
 portable_scrubber.ogg - https://freesound.org/people/Beethovenboy/sounds/384335/ - CC0 (by Beethovenboy)
 alarm.ogg - https://github.com/Baystation12/Baystation12/commit/41b11ef289bccfdfa2940480beb9c1e3f50c3b93, fire_alarm.ogg CC-BY-SA-3.0
 reclaimer_ambience.ogg - https://freesound.org/people/cmorris035/sounds/319152/, Woodchipper Start (heavy machine) 02.wav CC0 (by cmorris035)
+flesh_heart.ogg - Used from game LonaRPG
```
Comment (ficcialfaint) [inline Resources/Audio/Ambience/Objects/license.txt:11]: This has to contain a link and a license AND (!!) probably should be moved into an RGA (attributions.yml)

File: Resources/Audio/Animals/Flesh/license.txt [hunk 18]
```diff
@@ -0,0 +1,24 @@
...
+
+    flesh_worm_dead.ogg:
+
+    flesh_worm_scream.ogg:
+
+    spike_gun_reload.ogg:
+
+    spike_gun_shot.ogg:
+
+    throw_worm.ogg:
```
Comment (ficcialfaint) [inline Resources/Audio/Animals/Flesh/license.txt:23]: https://github.com/space-wizards/space-station-14/pull/16937#discussion_r1209601498

**Comment by VigersRay** [comment]
> Weren’t these sprites taken from another game? I remember you mentioning that during a discussion on Discord about them.

Yes, because someone has to redraw them. Although the author has given me permission to use them, I don't know if that permission is right for you.

File: Resources/Locale/en-US/game-ticking/game-presets/preset-flesh-cult-and-traitor.ftl [hunk 19]
```diff
@@ -0,0 +1,2 @@
+flesh-cult-and-traitor-title = Культ плоти и Предатели
+flesh-cult-and-traitor-description = На станции одновременно действует культ плоти и предатели.
```
Comment (ficcialfaint) [inline Resources/Locale/en-US/game-ticking/game-presets/preset-flesh-cult-and-traitor.ftl:2]: uh?

File: Resources/Textures/Objects/Weapons/Melee/flesh_fist.rsi/meta.json [hunk 20]
```diff
@@ -0,0 +1,22 @@
+{
+  "version": 1,
+  "license": "CC-BY-SA-3.0",
+  "copyright": "Created by Cthulhuwu (Discord)",
```
Comment (ficcialfaint) [inline Resources/Textures/Objects/Weapons/Melee/flesh_fist.rsi/meta.json:4]: discrim

File: Resources/Textures/Objects/Weapons/Melee/flesh_blade.rsi/meta.json [hunk 21]
```diff
@@ -0,0 +1,22 @@
+{
+  "version": 1,
+  "license": "CC-BY-SA-3.0",
+  "copyright": "Created by https://github.com/VigersRay and Cthulhuwu (Discord)",
```
Comment (ficcialfaint) [inline Resources/Textures/Objects/Weapons/Melee/flesh_blade.rsi/meta.json:4]: discrim

File: Resources/Textures/Objects/Weapons/Guns/Projectiles/flesh_toxic.rsi/meta.json [hunk 22]
```diff
@@ -0,0 +1,14 @@
+{
+  "version": 1,
+  "license": "CC-BY-NC-3.0",
```
Comment (ficcialfaint) [inline Resources/Textures/Objects/Weapons/Guns/Projectiles/flesh_toxic.rsi/meta.json:3]: any reason for it being `CC-BY-NC-3.0` and not `CC-BY-SA-3.0` as all other RSIs in this PR?

File: Resources/Textures/Objects/Tools/flesh_claw.rsi/meta.json [hunk 23]
```diff
@@ -0,0 +1,22 @@
+{
+  "version": 1,
+  "license": "CC-BY-SA-3.0",
+  "copyright": "Created by https://github.com/VigersRay and Cthulhuwu (Discord)",
```
Comment (ficcialfaint) [inline Resources/Textures/Objects/Tools/flesh_claw.rsi/meta.json:4]: discrim

File: Resources/Textures/Mobs/Aliens/FleshCult/flesh_cult_pudge.rsi/meta.json [hunk 24]
```diff
@@ -0,0 +1,56 @@
+﻿{
+    "version": 1,
+    "license": "CC-BY-SA-3.0",
+    "copyright": "",
```
Comment (ficcialfaint) [inline Resources/Textures/Mobs/Aliens/FleshCult/flesh_cult_pudge.rsi/meta.json:4]: empty

File: Resources/Textures/Mobs/Aliens/FleshCult/flesh_cult_abom.rsi/meta.json [hunk 25]
```diff
@@ -0,0 +1,56 @@
+{
+    "version": 1,
+    "license": "CC-BY-SA-3.0",
+    "copyright": "",
```
Comment (ficcialfaint) [inline Resources/Textures/Mobs/Aliens/FleshCult/flesh_cult_abom.rsi/meta.json:4]: empty

File: Resources/Textures/Objects/Weapons/Guns/Projectiles/flesh_toxic.rsi/meta.json (hunk 22 above)
Comment (VigersRay) [reply Resources/Textures/Objects/Weapons/Guns/Projectiles/flesh_toxic.rsi/meta.json:3]: i dont know, maybe typo

**Review by ficcialfaint — CHANGES_REQUESTED** [review]
Fix the failing tests

**Comment by moonheart08** [comment]
Major license issues, all assets need to be some Creative Commons license. "taken from X game" is not sufficient.

**Comment by VigersRay** [comment]
> Major license issues, all assets need to be some Creative Commons license. "taken from X game" is not sufficient.


What about the space_asshole license? I can ask permission from the sprite author if that's appropriate.
- files: ["space_asshole.ogg"]
  license: "Custom"

**Comment by moonheart08** [comment]
No.

**Pull request closed without merging**